### PR TITLE
Modify chat & player tests to ts

### PR DESCRIPTION
**Pull request by jasulcaf**
Related to Issue #36, this PR is to change Chat and Player tests to TS. Included I have also put both the .ts & .js version of the class declarations (for both Chat and Player) as I saw that the current tests passing (database) used .ts. Currently trying to see if all .ts will work, afterwards can delete the .js if it does.

### CODE REVIEW

**Comment by jasulcaf** [comment]
Not quite sure why karma & package-json file changed, maybe from when I tried to run the tests on my machine and I got stuck? Can revert those changes to be same as main if needed.

**Comment by gypark23** [comment]
chat.spec.ts compiles and their tests run. There are 12 tests total, and only 2 tests pass right now. I'm not sure if it is intended, but just letting you know. So technically chat.spec.test can be merged (without compile errors) if you are ok with failing tests.

NativeScript / 16.1 (16.1; iPhone) chat_tests Retrieving Player List for an empty Chat FAILED
        Expected: []
        Actual: []
            at Object.<anonymous> (file:///app/bundle.js:678:12)
            at runTest (eval at runTest (file:///app/vendor.js:52241:17), <anonymous>:2725:35)
            at Test.run (eval at runTest (file:///app/vendor.js:52241:17), <anonymous>:2708:9)
            at eval (eval at runTest (file:///app/vendor.js:52241:17), <anonymous>:2972:16)
            at processTaskQueue (eval at runTest (file:///app/vendor.js:52241:17), <anonymous>:2293:26)
            at eval (eval at runTest (file:///app/vendor.js:52241:17), <anonymous>:2297:13)

NativeScript / 16.1 (16.1; iPhone) chat_tests Inserting Player1 into Chat1 FAILED
        Died on test #1: Cannot read property 'push' of undefined
            at Module../src/tests/chat.spec.ts (file:///app/bundle.js:681:7)
            at __webpack_require__ (file:///app/runtime.js:23:42)
            at webpackContext (file:///app/bundle.js:879:9)
            at Array.map (<anonymous>)
            at runTests (file:///app/bundle.js:615:22)
            at file:///app/vendor.js:52311:63
            at Generator.next (<anonymous>)
        TypeError: Cannot read property 'push' of undefined
            at Player.insertChat (file:///app/bundle.js:410:25)
            at Chat.insertPlayer (file:///app/bundle.js:181:23)
            at Object.<anonymous> (file:///app/bundle.js:682:24)
            at runTest (eval at runTest (file:///app/vendor.js:52241:17), <anonymous>:2725:35)
            at Test.run (eval at runTest (file:///app/vendor.js:52241:17), <anonymous>:2708:9)
            at eval (eval at runTest (file:///app/vendor.js:52241:17), <anonymous>:2972:16)
            at processTaskQueue (eval at runTest (file:///app/vendor.js:52241:17), <anonymous>:2293:26)
            at eval (eval at runTest (file:///app/vendor.js:52241:17), <anonymous>:2297:13)

NativeScript / 16.1 (16.1; iPhone) chat_tests Player1 sends a message to Chat1 FAILED
        Died on test #1: Cannot read property 'length' of undefined
            at Module../src/tests/chat.spec.ts (file:///app/bundle.js:691:7)
            at __webpack_require__ (file:///app/runtime.js:23:42)
            at webpackContext (file:///app/bundle.js:879:9)
            at Array.map (<anonymous>)
            at runTests (file:///app/bundle.js:615:22)
            at file:///app/vendor.js:52311:63
            at Generator.next (<anonymous>)
        TypeError: Cannot read property 'length' of undefined
            at Player.getChat (file:///app/bundle.js:341:45)
            at Player.sendChatMessage (file:///app/bundle.js:394:30)
            at Object.<anonymous> (file:///app/bundle.js:692:26)
            at runTest (eval at runTest (file:///app/vendor.js:52241:17), <anonymous>:2725:35)
            at Test.run (eval at runTest (file:///app/vendor.js:52241:17), <anonymous>:2708:9)
            at eval (eval at runTest (file:///app/vendor.js:52241:17), <anonymous>:2972:16)
            at processTaskQueue (eval at runTest (file:///app/vendor.js:52241:17), <anonymous>:2293:26)
            at eval (eval at runTest (file:///app/vendor.js:52241:17), <anonymous>:2297:13)

NativeScript / 16.1 (16.1; iPhone) chat_tests Add Player2 into Chat 1 and see if can see the message previously sent FAILED
        Died on test #1: Cannot read property 'push' of undefined
            at Module../src/tests/chat.spec.ts (file:///app/bundle.js:697:7)
            at __webpack_require__ (file:///app/runtime.js:23:42)
            at webpackContext (file:///app/bundle.js:879:9)
            at Array.map (<anonymous>)
            at runTests (file:///app/bundle.js:615:22)
            at file:///app/vendor.js:52311:63
            at Generator.next (<anonymous>)
        TypeError: Cannot read property 'push' of undefined
            at Player.insertChat (file:///app/bundle.js:410:25)
            at Chat.insertPlayer (file:///app/bundle.js:181:23)
            at Object.<anonymous> (file:///app/bundle.js:698:24)
            at runTest (eval at runTest (file:///app/vendor.js:52241:17), <anonymous>:2725:35)
            at Test.run (eval at runTest (file:///app/vendor.js:52241:17), <anonymous>:2708:9)
            at eval (eval at runTest (file:///app/vendor.js:52241:17), <anonymous>:2972:16)
            at processTaskQueue (eval at runTest (file:///app/vendor.js:52241:17), <anonymous>:2293:26)
            at eval (eval at runTest (file:///app/vendor.js:52241:17), <anonymous>:2297:13)

NativeScript / 16.1 (16.1; iPhone) chat_tests Inserting messages into Chat2 FAILED
        Expected: 1
        Actual: 2
            at Object.<anonymous> (file:///app/bundle.js:711:12)
            at runTest (eval at runTest (file:///app/vendor.js:52241:17), <anonymous>:2725:35)
            at Test.run (eval at runTest (file:///app/vendor.js:52241:17), <anonymous>:2708:9)
            at eval (eval at runTest (file:///app/vendor.js:52241:17), <anonymous>:2972:16)
            at processTaskQueue (eval at runTest (file:///app/vendor.js:52241:17), <anonymous>:2293:26)
            at eval (eval at runTest (file:///app/vendor.js:52241:17), <anonymous>:2297:13)

        Died on test #3: Cannot read property 'getMessageContent' of undefined
            at Module../src/tests/chat.spec.ts (file:///app/bundle.js:708:7)
            at __webpack_require__ (file:///app/runtime.js:23:42)
            at webpackContext (file:///app/bundle.js:879:9)
            at Array.map (<anonymous>)
            at runTests (file:///app/bundle.js:615:22)
            at file:///app/vendor.js:52311:63
            at Generator.next (<anonymous>)
        TypeError: Cannot read property 'getMessageContent' of undefined
            at Object.<anonymous> (file:///app/bundle.js:712:30)
            at runTest (eval at runTest (file:///app/vendor.js:52241:17), <anonymous>:2725:35)
            at Test.run (eval at runTest (file:///app/vendor.js:52241:17), <anonymous>:2708:9)
            at eval (eval at runTest (file:///app/vendor.js:52241:17), <anonymous>:2972:16)
            at processTaskQueue (eval at runTest (file:///app/vendor.js:52241:17), <anonymous>:2293:26)
            at eval (eval at runTest (file:///app/vendor.js:52241:17), <anonymous>:2297:13)

NativeScript / 16.1 (16.1; iPhone) chat_tests Begin a Vote in Chat 1 with Player 1 & 2; and Killer 1 FAILED
        Died on test #1: Cannot read property 'push' of undefined
            at Module../src/tests/chat.spec.ts (file:///app/bundle.js:718:7)
            at __webpack_require__ (file:///app/runtime.js:23:42)
            at webpackContext (file:///app/bundle.js:879:9)
            at Array.map (<anonymous>)
            at runTests (file:///app/bundle.js:615:22)
            at file:///app/vendor.js:52311:63
            at Generator.next (<anonymous>)
        TypeError: Cannot read property 'push' of undefined
            at Killer.insertChat (file:///app/bundle.js:410:25)
            at Chat.insertPlayer (file:///app/bundle.js:181:23)
            at Object.<anonymous> (file:///app/bundle.js:719:11)
            at runTest (eval at runTest (file:///app/vendor.js:52241:17), <anonymous>:2725:35)
            at Test.run (eval at runTest (file:///app/vendor.js:52241:17), <anonymous>:2708:9)
            at eval (eval at runTest (file:///app/vendor.js:52241:17), <anonymous>:2972:16)
            at processTaskQueue (eval at runTest (file:///app/vendor.js:52241:17), <anonymous>:2293:26)
            at eval (eval at runTest (file:///app/vendor.js:52241:17), <anonymous>:2297:13)

File: appGeo/src/tests/player.spec.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+// QUnit.test("killer kills a player", assert => {
+//     //Check that the killPlayer() function succeeds
+//     assert.equal(killer1.killPlayer(1, playerMap.get_player_hash()), SUCCESS);
+//     //Check the number of Kills that Killer 1 commited
+//     assert.equal(killer1.getTotalKillCount(), 1);
+//     assert.equal(player1.getAliveStatus(), DEAD);
+//     //Check the Remaining Amount of Kills left
+//     assert.equal(killer1.getRemainingDailyKillCount(), (DAILYMAXKILLCOUNT - 1));
+// });
```
Comment (gypark23) [inline appGeo/src/tests/player.spec.ts:81]: These all needs to be commented out in order to have everything compiled.

**Review by gypark23 — CHANGES_REQUESTED** [review]
The files compile if you remove or comment out player.spec.ts. 
Not sure why it isnt't compiling since it seems to be free of error on vscode, but I am guessing there are many errors in js modules 
```
import {Chat} from '../app/chat/chat_class_declaration';
import {Player} from '../app/player/player_class_declaration';
import {Killer} from '../app/player/player_class_declaration';
```
that just aren't detected by vscode.

**Comment by gypark23** [comment]
I see the PR passed the workflow test. Is it good to go? @jasulcaf

**Comment by jasulcaf** [comment]
Yup, PR good to go now that tests all pass.

**Review by gypark23 — APPROVED** [review]

**Pull request merged**